### PR TITLE
model_collect: add services for module importation

### PR DESCRIPTION
**Pull request by Morriar**
Build a new hierarchy poset from MClass inheritance

Signed-off-by: Alexandre Terrasa <alexandre@moz-code.org>

### CODE REVIEW

**Comment by Morriar** [comment]
test this please

File: src/model/model_collect.nit [hunk 1]
```diff
@@ -220,6 +220,17 @@ redef class MClass
...
 	end
 
+	# Build the hierarchy poset for `self`
+	fun hierarchy_poset(mainmodule: MModule, view: ModelView): POSet[MClass] do
+		var mclasses = new HashSet[MClass]
+		mclasses.add self
+		mclasses.add_all collect_ancestors(view)
+		mclasses.add_all collect_parents(view)
+		mclasses.add_all collect_children(view)
+		mclasses.add_all collect_descendants(view)
```
Comment (privat) [inline src/model/model_collect.nit]: you do not need the four relationships. Only the two large one should be enough

File: src/model/model_collect.nit [hunk 2]
```diff
@@ -220,6 +220,17 @@ redef class MClass
 		return res
 	end
 
+	# Build the hierarchy poset for `self`
+	fun hierarchy_poset(mainmodule: MModule, view: ModelView): POSet[MClass] do
```
Comment (privat) [inline src/model/model_collect.nit]: The documentation is not clear on what exactly is returned.

**Comment by Morriar** [comment]
Reroll for @privat remarks

**Comment by Morriar** [comment]
test this please

**Pull request merged**